app/controllers/users.js

```diff
@@ -139,21 +147,20 @@ exports.addDonation = function(req, res) {
       User.findOne({
         _id: req.user._id
       })
-      .exec(function(err, user) {
-        // Confirm that this object hasn't already been entered
-        var duplicate = false;
-        for (var i = 0; i < user.donations.length; i++ ) {
-          if (user.donations[i].crowdrise_donation_id === req.body.crowdrise_donation_id) {
-            duplicate = true;
+        .exec(function (err, user) {
+          // Confirm that this object hasn't already been entered
+          var duplicate = false;
+          for (var i = 0; i < user.donations.length; i += 1) {
+            if (user.donations[i].crowdrise_donation_id === req.body.crowdrise_donation_id) {
+              duplicate = true;
```

Comment:
For performance sakes here, you should have an early break/return statement once this condition is met as there is no need going through the remainder of the list.

So you should have something like

```javascript
if (user.donations[i].crowdrise_donation_id === req.body.crowdrise_donation_id) {
    duplicate = true;
    break; // - out of the for loop
}
```

Or better still, you could use the convenient `Array.prototype.some` method and replace this entire block
```javascript
var duplicate = false;
for (var i = 0; i < user.donations.length; i += 1) {
    if (user.donations[i].crowdrise_donation_id === req.body.crowdrise_donation_id) {
        duplicate = true;
    }
}
```
with

```javascript
const duplicate  = user.donations.some(donation => donation.crowdrise_donation_id ===  req.body.crowdrise_donation_id);
```